City-of-Helsinki/hel-fi-drupal-grants
# fix: AU-1697: Changes to the "Search for application" search view

# [AU-1697](https://helsinkisolutionoffice.atlassian.net/browse/AU-1697)

## What was done
The "**Search for application**" search view contains a "**Show only the grants that can be applied for**" checkbox, which is supposed to filter out any applications that a user cannot apply for. This filter was previously broken, since it only checked the value of the `field_application_open` field, which was designed to indicates if the application is open or not, but in reality it only reflected the state of the referenced Webform.

Now, using the "**Show only the grants that can be applied for**" checkbox, the applications are filtered in the following way:

- `field_application_continuous` must be set to `TRUE`
-  **OR**
- The `current time` must fall within the time range specified in  the `field_application_period` field.

In addition to this, the `field_application_open` has been removed completely, since it was deemed unnecessary.

## How to install

Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-1697-node-application-open-fix`
  * `make fresh`
  * `make drush-cr`

## How to test
- Login with an admin user and re-index the site at: `/admin/config/search/search-api/index/search_index`.
- Now go to `/en/application-search` and search for applications with the "**Show only the grants that can be applied for**" checkbox checked. You should only be able the see applications that meet the criteria listed above.

- To test this more thoroughly, you can select a specific application and edit the applications form:

  - Set the form to be "**continuous**" and adjust the **application period** to be in the **future**. The form should show up when using the "**Show only the grants that can be applied for**" filter, since it is set to be "**continuous**".
  
  - You can also do the opposite: Set the form **not to be "continuous"**, and adjust the **application period** so that the form is set to open **in a few minutes**. Go back to the application search and use the "**Show only the grants that can be applied for**" filter, which should result in the application **not showing up**. Wait a few minutes until the **current time** is within the **application period** and search again. Now, the application **should show up.**

[AU-1697]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
